remove code deprecated after 0.7 release

ChitsimranGill opened the pull request:
### Summary
Removed deprecated code from estimator, sampler, and qiskit_runtime_service.

kt474 reviewed (commented):
There are a couple more deprecated parameters like `circuits` mentioned in https://github.com/Qiskit/qiskit-ibm-runtime/issues/807 that should also be removed.

You can take a look at the terra [PR](https://github.com/Qiskit/qiskit-terra/pull/9480) that removed these

ChitsimranGill commented:
Removed circuits, observables, and parameters args from sampler and estimator. Also, removed backend_name from runtime_options. 

Do we need to remove [auth](https://github.com/Qiskit/qiskit-ibm-runtime/blob/202d0219c67e327663777b971dbd74095cb35345/qiskit_ibm_runtime/qiskit_runtime_service.py#L161) arg from qiskit_runtime_service

kt474 commented:
> Do we need to remove [auth](https://github.com/Qiskit/qiskit-ibm-runtime/blob/202d0219c67e327663777b971dbd74095cb35345/qiskit_ibm_runtime/qiskit_runtime_service.py#L161) arg from qiskit_runtime_service?

Yeah we can remove the auth param - can you also add a release note that highlights all the removals

kt474 reviewed (commented):
Looks pretty good to me - we'll have to fix several integration tests but that can be done after finish https://github.com/Qiskit/qiskit-ibm-runtime/pull/782. I can take of it

ikkoham commented on qiskit_ibm_runtime/estimator.py:
```diff
@@ -99,32 +92,12 @@ class Estimator(BaseEstimator):
-        observables: Optional[Iterable[SparsePauliOp]] = None,
-        parameters: Optional[Iterable[Iterable[Parameter]]] = None,
-        service: Optional[QiskitRuntimeService] = None,
         session: Optional[Union[Session, str, IBMBackend]] = None,
```
IMO, these options should be keyword only arguments. But it might be preference.

ikkoham commented on qiskit_ibm_runtime/estimator.py:
```diff
@@ -138,71 +111,36 @@ def __init__(
-                )
             default_options = asdict(Options())
             self._options = Options._merge_options(default_options, options_copy)
             skip_transpilation = self._options.get("transpilation", {}).get(
```
Do we need this processing? (From line 132 to 137)

kt474 replied:
You're right, I don't believe this is necessary

ikkoham commented on qiskit_ibm_runtime/sampler.py:
```diff
@@ -113,66 +91,37 @@ def __init__(
-                )
             default_options = asdict(Options())
             self._options = Options._merge_options(default_options, options_copy)
             skip_transpilation = self._options.get("transpilation", {}).get(
```
Same comment with estimator.

ikkoham commented on qiskit_ibm_runtime/sampler.py:
```diff
@@ -78,28 +72,12 @@ class Sampler(BaseSampler):
-        circuits: Optional[Union[QuantumCircuit, Iterable[QuantumCircuit]]] = None,
-        parameters: Optional[Iterable[Iterable[Parameter]]] = None,
-        service: Optional[QiskitRuntimeService] = None,
         session: Optional[Union[Session, str, IBMBackend]] = None,
```
Same comment with Estimator.

ikkoham reviewed (commented):
Almost LGTM! I left some minor comments.

This PR must be released before Terra 0.24.0 release. (it breaks this library.)
